ignite/handlers/checkpoint.py

```diff
@@ -697,7 +697,10 @@ def reload_objects(self, to_load: Mapping, load_kwargs: Optional[Dict] = None, *
 
         checkpoint_fp = filename_pattern.format(**filename_dict)
 
-        path = self.save_handler.dirname / checkpoint_fp
+        if isinstance(self.save_handler, DiskSaver):
```

Comment:
Raise a TypeError in the begining of this function if `self.save_handler` is not a DiskSaver